weblab/entities/tests/test_views.py

```diff
@@ -416,6 +417,175 @@ def test_returns_404_if_commit_not_found(self, client, logged_in_user):
+            '/entities/models/compare/%s/%s/info' % (v1_spec, v2_spec)
+        )
+
+        assert response.status_code == 200
+        data = json.loads(response.content.decode())
+        versions = data['getEntityInfos']['entities']
+        assert versions[0]['id'] == v1.hexsha
+        assert versions[1]['id'] == v2.hexsha
+        assert versions[0]['author'] == model.author.full_name
+        assert versions[0]['visibility'] == 'public'
+        assert versions[0]['name'] == model.name
+        assert versions[0]['version'] == v1.hexsha
+        assert versions[0]['numFiles'] == 1
+        assert versions[0]['commitMessage'] == v1.message
+
+    def test_cannot_compare_entities_with_no_access(self, client, helpers):
+        model = recipes.model.make()
+        v1 = helpers.add_version(model, visibility='public')
+        v2 = helpers.add_version(model, visibility='private')
+
+        v1_spec = '%d:%s' % (model.pk, v1.hexsha)
+        v2_spec = '%d:%s' % (model.pk, v2.hexsha)
+        response = client.get(
+            '/entities/models/compare/%s/%s/info' % (v1_spec, v2_spec)
+        )
+
+        assert response.status_code == 200
+        data = json.loads(response.content.decode())
+        versions = data['getEntityInfos']['entities']
+        assert len(versions) == 1
+        assert versions[0]['id'] == v1.hexsha
+
+    def test_can_compare_entities_if_collaborator(self, client, logged_in_user, helpers):
+        model = recipes.model.make()
+        v1 = helpers.add_version(model, visibility='public')
+        v2 = helpers.add_version(model, visibility='private')
+        assign_perm('edit_entity', logged_in_user, model)
+
+        v1_spec = '%d:%s' % (model.pk, v1.hexsha)
+        v2_spec = '%d:%s' % (model.pk, v2.hexsha)
+        response = client.get(
+            '/entities/models/compare/%s/%s/info' % (v1_spec, v2_spec)
+        )
+
+        assert response.status_code == 200
+        data = json.loads(response.content.decode())
+        versions = data['getEntityInfos']['entities']
+        assert len(versions) == 2
+        assert versions[0]['id'] == v1.hexsha
+        assert versions[1]['id'] == v2.hexsha
+
+    def test_file_json(self, client, helpers):
+        model = recipes.model.make()
+        v1 = helpers.add_version(model, visibility='public')
+
+        v1_spec = '%d:%s' % (model.pk, v1.hexsha)
+        response = client.get(
+            '/entities/models/compare/%s/info' % v1_spec
+        )
+
+        assert response.status_code == 200
+        data = json.loads(response.content.decode())
+        versions = data['getEntityInfos']['entities']
+        assert versions[0]['id'] == v1.hexsha
+        assert versions[0]['numFiles'] == 1
+        file_ = versions[0]['files'][0]
+        assert file_['id'] == 'file1.txt'
+        assert file_['name'] == 'file1.txt'
+        assert file_['author'] == model.author.full_name
+        assert file_['filetype'] == 'TXTPROTOCOL'
+        #assert file_['masterFile']
```

Comment:
Why is this commented out?

Reply:
It should've been removed - I copy pasted it from another bit of code that did need to serve up that field, but it looks as if this one doesn't. I'll remove the line.